projects/templates/projects/view_project.html

```diff
@@ -133,43 +133,92 @@ <h3>Project Users</h3>
 </form>
 {% endif %}
 
-<h4>Current Users</h4>
-<ul class="list-unstyled">
+<br>
+<br>
+<table width=100%>
+  <tr>
+    <th>Username</th>
+    <th>First Name</th>
+    <th>Last Name</th>
+    <th>Email</th>
+    <th>Role</th>
+    <th></th>
+  </tr>
   {% for u in users %}
-  <li>
-      {% if is_pi and u.username != request.user.username %}
-      <form role="form" method="post" style="display:inline">
-        {% csrf_token %}
-        <input type="hidden" name="user_ref" value="{{ u.username }}">
-        <button type="submit" class="btn btn-xs btn-link" name="del_user"><i class="fa fa-minus-square text-danger"></i><span class="sr-only">Remove user</span></button>
-      </form>
-    {% endif %}
-    <strong>{{ u.username }}</strong>{% if u.email %}: {{ u.first_name }} {{ u.last_name }} ({{ u.email }}){% endif %}
-  </li>
+  <tr>
+    <td>{{ u.username }}</td>
+    <td>{{ u.first_name }}</td>
+    <td>{{ u.last_name }}</td>
+    <td>{{ u.email }}</td>
+    <td>
+       <form role="form" method="post" style="display:inline">
+         {% csrf_token %}
+         <input type="hidden" name="user_ref" value="{{ u.username }}">
+         <select id="role_{{ u.email }}" name="user_role" class="form-select role_selector" disabled>
+           {% for role in roles %}
+             <option value="{{ role }}" {% if role == u.role %}selected disabled{% endif %}>{{ role }}</option>
+           {% endfor %}
+         </select>
+         <button type="submit" id="submit_role_{{ u.email }}" class="btn btn-xs btn-primary" name="change_role" style="display:none;">Submit</button>
+         <button type="button" onclick="resetRole('role_{{ u.email }}')" id="cancel_role_{{ u.email }}" class="btn btn-xs btn-danger" style="display:none;">Cancel</button>
+       </form>
+    </td>
+    <td>
+      {% if u.username != request.user.username %}
+      <div class="dropdown">
+        <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" {% if not can_manage_project and not can_manage_project_membership %}disabled{%endif%}>
+          <i class="fa fa-gear text-info dropbtn"></i><span class="sr-only">Manager Member</span>
+        </button>
+        <div class="dropdown-menu dropdown-content" aria-labelledby="dropdownMenuButton">
+          {% if can_manage_project %}
+          <button onclick="enableChangeRole('{{ u.email }}')" class="btn btn-xs btn-block">Change Role</button>
```

Comment:
Related to the above, how can somebody manage the project but not the membership? It would seem that either they can manage membership, in which case we should the button and form, or they can't, in which case we should show neither.

Reply:
admins "can manage project", including membership. managers "can only manage membership". (i.e. they can only add/remove users, but can't change roles of the members.) that is to say, only PI can promote/demote members.